server/src/food/handler/FoodItem.handler.ts

```diff
@@ -0,0 +1,127 @@
+import { NextFunction, Request, Response } from "express";
+import { Types } from "mongoose";
+import { DbError, ERRORS } from "../../app/index";
+import { DbExtension } from "../../db/interface/db-extension";
+import { FoodItem } from "../index";
+
+export class FoodItemHandler {
+
+  constructor (private _db: DbExtension) {}
+
+  public getAll = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      let result;
+
+      if (req.query.hasOwnProperty("socobouserid")) {
+        const socoboUserId = req.query.socobouserid;
+        await this._checkIfSocoboUserExists(socoboUserId, "getAll(..)");
+
+        result = (req.query.hasOwnProperty("resolve"))
+          ? result = await this._db.fooditem.getAllBySocoboUserIdResolved(socoboUserId, false)
+          : (req.query.hasOwnProperty("resolve-deep"))
+            ? await this._db.fooditem.getAllBySocoboUserIdResolved(socoboUserId, true)
+            : await this._db.fooditem.getAllBySocoboUserId(socoboUserId);
+
+      } else {
+
+        result = (req.query.hasOwnProperty("resolve"))
+          ? await this._db.fooditem.getAllResolved(false)
+          : (req.query.hasOwnProperty("resolve-deep"))
+            ? await this._db.fooditem.getAllResolved(true)
+            : await this._db.fooditem.getAll();
+      }
+
+      res.status(200).json(result);
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  public getById = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+
+      const id = new Types.ObjectId(req.params.id);
+      const result = (req.query.hasOwnProperty("resolve"))
+        ? await this._db.fooditem.getByIdResolved(id, false)
+        : (req.query.hasOwnProperty("resolve-deep"))
+          ? await this._db.fooditem.getByIdResolved(id, true)
+          : await this._db.fooditem.getById(id);
+
+      res.status(200).json(result);
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  public save = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const item = new FoodItem().clone(req.body);
+
+      await this._checkIfSocoboUserExists(item.socoboUserId.toHexString(), "save(..)");
+      await this._checkIfFoodItemTemplateExists(item.foodItemTemplateId, "save(..)");
+
+      const result = await this._db.fooditem.save(item) as Types.ObjectId;
+      res.status(201).json(item.setId(result));
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  public updateById = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const foodItemId = new Types.ObjectId(req.params.id);
+      const socoboUserId = new Types.ObjectId(req.body.socoboUserId);
+      const foodItemTemplateId = new Types.ObjectId(req.body.foodItemTemplateId);
+
+      await this._checkIfSocoboUserExists(socoboUserId.toHexString(), "updateById(..)");
```

Comment:
Just a question because I forgot it or just don't know it. Why do you use a hex string here?

Reply:
Because my inconsistent implementation. SocoboUser repository works with object id as string and toHexString(..) produce this string from Types.ObjectId. 

But if I look I doesn't need this. Thanks for pointing. I will change this too.